04.ls/ls.rb

```diff
@@ -1,15 +1,24 @@
 #!/usr/bin/env ruby
 # frozen_string_literal: true
 
+require 'optparse'
+
 COLUMN_COUNT = 3
 
 def main
-  files = Dir.glob('*').sort_by(&:downcase)
+  params = ARGV.getopts('a')
+  files = prepared_files(params)
   row_count = files.count.ceildiv(COLUMN_COUNT)
   columns = split_into_columns(files, COLUMN_COUNT, row_count)
   output(columns, row_count)
 end
 
+# -rオプションもprepared_filesメソッドに実装予定
+def prepared_files(params)
+  raw_files = Dir.glob('*', params['a'] ? File::FNM_DOTMATCH : 0)
```

Comment:
```suggestion
  hoge = params['a'] ? File::FNM_DOTMATCH : 0
  raw_files = Dir.glob('*', hoge)
```

引数の中に式が埋め込まれていると読みづらいので一度変数に入れた方がいいかも
https://x.com/jnchito/status/1326468865103073280

Reply:
Suggestionありがとうございます、修正いたしました。
こちら、自分なりに短く出来て日報で「上手く出来て感動した！」みたいに書いていたので、短く出来ても結局人間にとって読みやすくないと意味がないと理解し反省しました😅（そしてちょっと恥ずかしい）

日報：
![image](https://github.com/user-attachments/assets/9eba026c-4059-4fde-bec5-a4fcff35392d)